### PR TITLE
Optimize helpers to handle amps from generic machine driver

### PR DESCRIPTION
**Pull request by gongjun-song**
sof_sdw_rt1308.c/sof_sdw_rt1316.c/sof_sdw_rt1318.c handle amp in basically the same way, optimized and merged into one file.

Signed-off-by: Gongjun Song <gongjun.song@intel.com>

### CODE REVIEW

**Comment by bardliao** [comment]
SOFCI TEST

**Review by bardliao — DISMISSED** [review]

File: sound/soc/intel/boards/sof_sdw_rt_amp.c [hunk 1]
```diff
@@ -31,36 +38,62 @@ static const struct snd_soc_dapm_route rt1308_map[] = {
...
+	/* get the codec name */
+	snprintf(codec_name, CODEC_NAME_SIZE, "%s", dai_name);
+
+	/* choose the right codec's map  */
+	if (strcmp(codec_name, "rt1308") == 0)
+		rt_amp_map = rt1308_map;
+	else if (strcmp(codec_name, "rt1316") == 0)
+		rt_amp_map = rt1316_map;
+	else
+		rt_amp_map = rt1318_map;
```
Comment (ranj063) [inline sound/soc/intel/boards/sof_sdw_rt_amp.c]: looks good except this part can be probably also move into a helper function and avoid duplication

**Review by ranj063 — DISMISSED** [review]

File: sound/soc/intel/boards/sof_sdw_rt_amp.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw_rt_amp.c]: agree with @ranj063

File: sound/soc/intel/boards/Makefile [hunk 2]
```diff
@@ -37,8 +37,7 @@ snd-soc-sof_da7219_max98373-objs := sof_da7219_max98373.o
 snd-soc-ehl-rt5660-objs := ehl_rt5660.o
 snd-soc-sof-ssp-amp-objs := sof_ssp_amp.o
 snd-soc-sof-sdw-objs += sof_sdw.o				\
-			sof_sdw_max98373.o sof_sdw_rt1308.o	\
-			sof_sdw_rt1316.o sof_sdw_rt1318.o	\
+			sof_sdw_max98373.o sof_sdw_rt_amp.o	\
```
Comment (plbossart) [inline sound/soc/intel/boards/Makefile:40]: commit title:

ASoC: Intel: sof_sdw: use common helpers for all Realtek amps

**Review by plbossart — CHANGES_REQUESTED** [review]

File: sound/soc/intel/boards/sof_sdw_rt_amp.c (hunk 1 above)
Comment (gongjun-song) [reply sound/soc/intel/boards/sof_sdw_rt_amp.c]: Thank @ranj063 and @plbossart , I have updated commit.

File: sound/soc/intel/boards/Makefile (hunk 2 above)
Comment (gongjun-song) [reply sound/soc/intel/boards/Makefile:40]: Updated

File: sound/soc/intel/boards/sof_sdw_rt_amp.c [hunk 3]
```diff
@@ -31,36 +38,69 @@ static const struct snd_soc_dapm_route rt1308_map[] = {
...
+};
+
+static const struct snd_soc_dapm_route rt1318_map[] = {
+	{ "Speaker", NULL, "rt1318-1 SPOL" },
+	{ "Speaker", NULL, "rt1318-1 SPOR" },
+	{ "Speaker", NULL, "rt1318-2 SPOL" },
+	{ "Speaker", NULL, "rt1318-2 SPOR" },
 };
 
+static const struct snd_soc_dapm_route *get_amp_route(struct snd_soc_pcm_runtime *rtd,
```
Comment (bardliao) [inline sound/soc/intel/boards/sof_sdw_rt_amp.c]: Maybe rename to `get_codec_name_and_route`?

Comment (gongjun-song) [reply sound/soc/intel/boards/sof_sdw_rt_amp.c]: thank @bardliao , updated

**Review by plbossart — DISMISSED** [review]

**Review by ranj063 — DISMISSED** [review]

**Review by plbossart — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

**Pull request merged**